### PR TITLE
Feature/basic gallery

### PR DESCRIPTION
**Pull request by salesfelipe**
#### What is the purpose of this pull request?

The first version of gallery canonical component 

#### What problem is this solving?

Displays an orderable list of objects to the user 

#### How should this be manually tested?

[workspace](https://gallery--storecomponents.myvtex.com/_v/gallery)

#### Screenshots or example usage

image

#### Types of changes

* [ ] Bug fix (non-breaking change which fixes an issue)
* [x] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to change)
* [ ] Requires change to documentation, which has been updated accordingly.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+
+<!--- What is the motivation and context for this change? -->
+
+#### How should this be manually tested?
+
+#### Screenshots or example usage
+
+#### Types of changes
+
+* [ ] Bug fix (non-breaking change which fixes an issue)
```
Comment (andreldsa) [inline .github/PULL_REQUEST_TEMPLATE.md]: a non-breaking

Comment (salesfelipe) [reply .github/PULL_REQUEST_TEMPLATE.md]: 👍

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+* [ ] New feature (non-breaking change which adds functionality)
```
[the remark this answers is not in the document]
Comment (salesfelipe) [reply .github/PULL_REQUEST_TEMPLATE.md]: 👍

File: react/Gallery.js [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+  products: GalleryContent.propTypes.productList,
+  /** Query used to find the products */
+  search: PropTypes.string.isRequired,
+}
+
+Gallery.defaultProps = {
+  columnsQuantityLarge: 5,
+  columnsQuantityMedium: 3,
+  products: products,
+  search: 'Celulares',
```
[the remark this answers is not in the document]
Comment (salesfelipe) [reply react/Gallery.js]: 👍

File: pages/pages.json [hunk 4]
```diff
@@ -0,0 +1,13 @@
+{
+  "pages": {
+    "gallery": {
+      "path": "/gallery"
```
Comment (brunojdo) [inline pages/pages.json]: `_v/gallery`

File: react/Gallery.js [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+  'sortBy.higherPrice',
+  'sortBy.lowerPrice',
+  'sortBy.nameAZ',
+  'sortBy.nameZA',
+]
+
+/**
+ * Canonical gallery that displays a list of given products.
+ */
+class Gallery extends React.Component {
```
Comment (brunojdo) [inline react/Gallery.js]: use  import { Component } ...

File: react/Gallery.js [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    columnsQuantityMedium: {
+      title: 'Columns quantity (Medium Viewport)',
+      type: 'number',
+      enum: [2, 3],
+      default: 3,
+    },
+  },
+}
+
+export default Gallery
```
Comment (brunojdo) [inline react/Gallery.js]: put this into class line

File: react/Gallery.js [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+          onSortChange={this.handleSortChange}
+          sortingOptions={sortingOptions}
+        />
+        <GalleryContent productList={this.state.products} {...this.props} />
+      </div>
+    )
+  }
+}
+
+Gallery.propTypes = {
```
Comment (brunojdo) [inline react/Gallery.js]: move into the class

File: react/components/GalleryContent.js [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+GalleryContent.propTypes = {
+  /** Quantity of columns when the viewport is large.*/
+  columnsQuantityLarge: PropTypes.number.isRequired,
+  /** Quantity of columns when the viewport is medium.*/
+  columnsQuantityMedium: PropTypes.number.isRequired,
+  /** Products to be displayed */
+  productList: PropTypes.arrayOf(GalleryItem.propTypes.item),
+}
+
+export default GalleryContent
```
Comment (brunojdo) [inline react/components/GalleryContent.js]: idem

File: react/components/GalleryHeader.js [hunk 9]
```diff
@@ -0,0 +1,79 @@
...
+import PropTypes from 'prop-types'
+import Dropdown from '@vtex/styleguide/lib/Dropdown'
+import VTEXClasses from '../utils/css-classes'
+
+import { FormattedMessage, injectIntl } from 'react-intl'
+
+/**
+ * Displays the header information of the Gallery: the query used to find the products and a selector to choose the sorting type.
+ */
+class GalleryHeader extends React.Component {
```
Comment (brunojdo) [inline react/components/GalleryHeader.js]: Destructure this

File: react/components/GalleryItem.js [hunk 10]
```diff
@@ -0,0 +1,73 @@
...
+              ListPrice: PropTypes.number.isRequired,
+            }),
+          }),
+        ).isRequired,
+      }),
+    ).isRequired,
+  }),
+}
+
+export default GalleryItem
```
Comment (brunojdo) [inline react/components/GalleryItem.js]: idem

File: .gitignore [hunk 11]
```diff
@@ -1,38 +1,36 @@
...
-.eslintrc
-.idea/
+# App specific
+node_modules/
+docs/_book/
+.tmp
+.idea
+npm-debug.log
+.build/
+lib
```
[the remark this answers is not in the document]
Comment (salesfelipe) [reply .gitignore]: 👍

File: manifest.json [hunk 12]
```diff
@@ -1,18 +1,16 @@
...
-  "description": "",
-  "mustUpdateAt": "2017-09-26",
-  "categories": [],
-  "settingsSchema": {},
-  "dependencies": {
-    "vtex.render-runtime": "0.x",
-    "vtex.render-builder": "0.x",
-    "vtex.react-slick": "0.x"
-  },
+  "version": "1.0.0",
```
[the remark this answers is not in the document]
Comment (salesfelipe) [reply manifest.json]: 👍

File: pages/pages.json (hunk 4 above)
Comment (salesfelipe) [reply pages/pages.json]: 👍

File: react/components/GalleryItem.js [hunk 13]
```diff
@@ -0,0 +1,73 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import { ExtensionPoint } from 'render'
+
+/**
+ * Normalizes the item received in the props to adapt to the extension point prop.
+ */
+class GalleryItem extends React.Component {
+  normalizeProductSummary(product) {
```
Comment (ClaudivanFilho) [inline react/components/GalleryItem.js]: Tu tem que mudar isso aqui @salesfelipe o product summary não mais aceita essas props https://github.com/vtex-apps/shelf/blob/master/react/ShelfItem.js#L11

File: react/Gallery.js (hunk 5 above)
Comment (salesfelipe) [reply react/Gallery.js]: 👍

File: react/Gallery.js (hunk 7 above)
Comment (salesfelipe) [reply react/Gallery.js]: 👍

File: react/components/GalleryItem.js (hunk 13 above)
Comment (salesfelipe) [reply react/components/GalleryItem.js]: 👍

**Pull request merged**